src/components/Tabs/ScatterPlotTab.tsx

```diff
@@ -765,13 +754,15 @@ export default memo(function ScatterPlotTab(props: ScatterPlotTabProps): ReactEl
           label={"X"}
           selected={xAxisFeatureName || ""}
           items={menuItems}
-          onChange={setXAxisFeatureName}
+          onChange={(key) => props.updateScatterPlotConfig({ xAxis: key })}
```

Comment:
I like the refactoring of this handler to directly call `props.updateScatterPlotConfig` with the new key!